src/pytest_workflow/file_tests.py

```diff
@@ -148,3 +159,45 @@ def repr_failure(self, excinfo, style=None):
             f"Observed md5sum '{self.observed_md5sum}' not equal to expected "
             f"md5sum '{self.expected_md5sum}' for file '{self.filepath}'"
         )
+
+
+class FileDiff(pytest.Item):
+    def __init__(self, parent: pytest.Collector, filepath: Path,
+                 expected_path: Path, workflow: Workflow):
+        """
+        Create a test for the file diff.
+        :param parent: The collector that started this item
+        :param filepath: The path to the file
+        :param expected_path: The path to diff filepath against
+        :param workflow: The workflow running to generate the file
+        """
+        name = "diff"
+        super().__init__(name, parent)
+        self.filepath = filepath
+        self.expected_path = expected_path
+        self.observed_diff = None
+        self.workflow = workflow
+
+    def runtest(self):
+        # Wait for the workflow to finish before we check the diff of a file.
+        self.workflow.wait()
+        if not self.workflow.matching_exitcode():
+            pytest.skip(f"'{self.parent.workflow.name}' did not exit with"
+                        f"desired exit code.")
+        # can't do the diff if either file is missing
+        if not self.expected_path.exists():
+            pytest.fail(f"'{self.expected_path}' does not exist.")
+        if not self.filepath.exists():
+            pytest.skip(f"'{self.filepath}' does not exist.")
+
+        self.observed_diff = file_diff(
+            self.expected_path, self.filepath,
+            encoding=self.parent.filetest.encoding
+        )
+        assert self.observed_diff is None
```

Comment:
assert not self.observed_diff

Same effect, equally clear and you can return strings from the file_diff.

Reply:
fixed in 476eb831688aa7a6ecd52e2e39299f0eb99de243